neqo-transport/src/frame.rs

```diff
@@ -442,17 +445,25 @@ impl<'a> Frame<'a> {
                 }
 
                 // Now check for the values for ACK_ECN.
-                if t == FRAME_TYPE_ACK_ECN {
-                    dv(dec)?;
-                    dv(dec)?;
-                    dv(dec)?;
-                }
+                let ecn_count: EcnCount = match t {
```

Comment:
There is enough code here that I want to suggest that you extract this arm of the match statement into a `decode_ack()` method.